editor/docks/editor_dock_manager.cpp

```diff
@@ -1187,17 +1158,40 @@ void DockContextPopup::_dock_select_draw() {
 	dock_select->draw_rect(center_panel_draw_rect, unusable_dock_color);
 
 	// Draw all dock slots.
-	for (int i = 0; i < EditorDockManager::DOCK_SLOT_MAX; i++) {
+	for (int i = 0; i < DockConstants::DOCK_SLOT_MAX; i++) {
+		int max_tabs = (i == DockConstants::DOCK_SLOT_BOTTOM) ? 6 : 3;
+		const EditorDockManager::DockSlot &dock_slot = dock_manager->dock_slots[i];
+
 		Rect2 dock_slot_draw_rect = dock_select_rects[i].grow_individual(-dock_spacing, -dock_top_spacing, -dock_spacing, -dock_spacing);
 		real_t tab_width = Math::round(dock_slot_draw_rect.size.width / max_tabs);
 		Rect2 tab_draw_rect = Rect2(dock_slot_draw_rect.position.x, dock_select_rects[i].position.y, tab_width - tab_spacing, tab_height);
+
+		real_t max_width = tab_width * max_tabs;
+		// Tabs may not fit perfectly, so they need to be re-centered.
+		if (max_width > dock_slot_draw_rect.size.x) {
+			tab_draw_rect.position.x -= int(max_width - dock_slot_draw_rect.size.x) / 2 * rtl_dir;
+		}
 		if (dock_select->is_layout_rtl()) {
 			tab_draw_rect.position.x += dock_slot_draw_rect.size.x - tab_draw_rect.size.x;
 		}
-		bool is_context_dock = context_tab_container == dock_manager->dock_slot[i];
-		int tabs_to_draw = MIN(max_tabs, dock_manager->dock_slot[i]->get_tab_count());
 
-		if (i == dock_select_rect_over_idx) {
+		int tab_count = 0;
+		{
+			int full_tab_count = dock_slot.container->get_tab_count();
+			for (int j = 0; j < full_tab_count; j++) {
+				if (!dock_slot.container->is_tab_hidden(j)) {
+					tab_count++;
+				}
+			}
+		}
+		int tabs_to_draw = MIN(max_tabs, tab_count);
+
+		bool is_context_dock = context_tab_container == dock_slot.container;
+		if (i == context_dock->dock_slot_index) {
```

Comment:
If any dock is dragged to a new slot, then the previous slot it was at is highlighted, instead of the current. Not sure where, but it seems that something isn't being updated right.

<img width="269" height="189" alt="image" src="https://github.com/user-attachments/assets/db48add5-3d0f-4832-a403-c1669359c057" />